app/src/main/java/zapsolutions/zap/baseClasses/BaseScannerActivity.java

```diff
@@ -170,8 +171,20 @@ public void showCameraWithPermissionRequest() {
 
     public void handleCameraResult(Result result) {
         if (result != null) {
-            ZapLog.debug(LOG_TAG, result.getContents());
+            ZapLog.debug(LOG_TAG, "Scanned content: " + result.getContents());
         }
+
+        // Note:
+        // * Wait 2 seconds to resume the preview.
+        // * On older devices continuously stopping and resuming camera preview can result in freezing the app.
+        // * I don't know why this is the case but I don't have the time to figure out.
+        Handler handler = new Handler();
+        handler.postDelayed(new Runnable() {
```

Comment:
We may want to make the Handler a member variable to the class, so we can stop the callbacks if the activity gets stopped. See: https://stackoverflow.com/a/23539589/12290689

Otherwise the callback could be invoked after the view is already destroyed and we get a null pointer here.